### PR TITLE
feat: Porting and refactoring the package

### PR DESCRIPTION
**Pull request by peterkulko**
**Description**
Porting and refactoring the [mfe-overrides](https://github.com/monteri/mfe-overrides) package.

- Code from [the private repository](https://github.com/monteri/mfe-overrides) is successfully migrated to rg-mfe-overrides.
- CI/CD checks are implemented and passing.
- Linters are configured and enforced.
- A detailed README.md with examples and usage instructions is written.
- A changelog (RG_CHANGELOG.rst) is created and populated with the initial changes.

[YouTrack](https://youtrack.raccoongang.com/issue/RGInt-313)

image

### CODE REVIEW

File: bin/mfe-cli.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        chalk.red.bold(
+          'Error: Git URL is required. Usage: mfe-cli install-frontend-build <git_repo_url>',
+        ),
+      );
+      process.exit(1);
+    }
+    installFrontendBuild(argument);
+    break;
+
+  case 'help':
```
Comment (peterkulko) [inline bin/mfe-cli.js:65]: Add more detailed description for each CLI commands (e.g. "Use this command in this dir")

File: lib/apply-overrides.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    files.forEach(async (file) => {
+      const relativePath = pathModule.relative(sourcePath, file);
+      const targetFilePath = pathModule.resolve(currentDirectory, relativePath);
+
+      // Create required directories if they don't exist
+      await fs.mkdir(pathModule.dirname(targetFilePath), { recursive: true });
+
+      // Copy only if the file is not in the ignored list
+      if (!IGNORED_FILES.includes(relativePath)) {
+        await fs.copyFile(file, targetFilePath, fs?.constants?.COPYFILE_FICLONE);
```
Comment (peterkulko) [inline lib/apply-overrides.js]: Remove `fs?.constants?.COPYFILE_FICLONE`

File: .eslint.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+const { createConfig } = require("@openedx/frontend-build");
```
Comment (peterkulko) [inline .eslint.js]:
```suggestion
const { createConfig } = require('@openedx/frontend-build');
```

File: package.json [hunk 4]
```diff
@@ -2,8 +2,21 @@
...
+  },
+  "license": "ISC",
+  "dependencies": {
+    "chalk": "^4.1.2",
+    "simple-git": "^3.20.0"
   },
-  "license": "ISC"
+  "devDependencies": {
+    "@openedx/frontend-build": "13.0.27",
+    "jest": "29.7.0"
```
Comment (peterkulko) [inline package.json]: Remove packages and commands related to tests

File: README.md [hunk 5]
```diff
@@ -1 +1,93 @@
...
+#### **Example**:
+```bash
+mfe-cli list-changes
+```
+
+---
+
+### **3. Apply Overrides**
+
+Copies or creates files from the `frontend-build/overrides` directory back into the working directory of your MFE.
```
Comment (ihor-romaniuk) [inline README.md]:
```suggestion
Moves modified or newly added files from the `frontend-build/overrides` directory back into the working directory of your MFE.
```

File: package.json [hunk 6]
```diff
@@ -2,8 +2,27 @@
...
   "version": "1.0.0",
   "main": "index.js",
+  "bin": {
+    "mfe-cli": "./bin/mfe-cli.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/raccoongang/rg-mfe-overrides"
+  },
+  "author": "Raccoon gang",
```
Comment (ihor-romaniuk) [inline package.json]:
```suggestion
  "author": "Raccoon Gang",
```

**Comment by ihor-romaniuk** [comment]
@PKulkoRaccoonGang 
- [nit] rename file [RG_CHANGELOG.rst RG_CHANGELOG.rst RG_CHANGELOG.rst](https://github.com/raccoongang/rg-mfe-overrides/blob/8cc65de068d373d46ff0771185117bb687ac09ca/RG_CHANGELOG.rst%0ARG_CHANGELOG.rst%0ARG_CHANGELOG.rst) to `RG_CHANGELOG.rst`
- [nit] add one more `=` under the `Added:` line in the file `RG_CHANGELOG.rst` for correct rst format
```
Added:
======
```

File: README.md (hunk 5 above)
Comment (peterkulko) [reply README.md]: I agree, it looks better

File: package.json (hunk 6 above)
Comment (peterkulko) [reply package.json]: Corrected

**Comment by peterkulko** [comment]
@ihor-romaniuk nice catch! Added new changes in the following commit https://github.com/raccoongang/rg-mfe-overrides/pull/1/commits/d3c2b4edf1cd9b7fb69bb40ea78b1232612ecf1d

**Review by ihor-romaniuk — APPROVED** [review]

**Pull request merged**